Zukini/EnumerableExtensions.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+
+namespace Zukini
+{
+    public static class EnumerableExtensions
+    {
+        /// <summary>
+        /// Shuffles items in Enumerable.
+        /// </summary>
+        /// <typeparam name="TSource"></typeparam>
+        /// <param name="source"></param>
+        /// <returns>IEnumerable</returns>
+        public static IEnumerable<TSource> Shuffle<TSource>(this IEnumerable<TSource> source)
+        {
+            return source.OrderBy(i => Guid.NewGuid());
+        }
+
+        /// <summary>
+        /// Retrieves random first item or throws error if item is not present.
+        /// </summary>
+        /// <typeparam name="TSource"></typeparam>
+        /// <param name="source"></param>
+        /// <param name="errorMessage"></param>
+        /// <returns>TSource</returns>
+        public static TSource RandomFirst<TSource>(this IEnumerable<TSource> source,
+            string errorMessage = "Cannot get random item: source is empty")
+        {
+            return source.Shuffle().FirstOrError(errorMessage);
+        }
+
+        /// <summary>
+        /// Returns first item or error if item is not present.
+        /// </summary>
+        /// <typeparam name="TSource"></typeparam>
+        /// <param name="source"></param>
+        /// <param name="errorMessage"></param>
+        /// <returns>TSource</returns>
+        public static TSource FirstOrError<TSource>(this IEnumerable<TSource> source, string errorMessage = "Item is not present.")
```

Comment:
Suggestion on the naming of the method here. In Linq - the convention is First() which throws an exception if there are none, or FirstOrDefault() if you want to return either the first element or null. I would propose you name it First(this IEnumerable<TSource>, string errorMessage = "Item is not present.") and have it behave like other standard linq methods.

Reply:
Fixed